src/models/loader.py

```diff
@@ -0,0 +1,50 @@
+import os
+from typing import Union, Literal
+from pydantic import BaseModel, Field
+from src.constants import ENV_FILE_PATH
+
+
+class FileLoader(BaseModel):
+    loader_name: Literal["file"] = Field(
+        description="The name of the loader.",
+    )
+    use_llama_parse: bool = Field(
+        default=False,
+        description="Use Llama Parser to parse the file.",
+    )
+    llama_cloud_api_key: str | None = Field(
+        default=None,
+        description="API key for Llama Parse.",
+    )
+
+    def __init__(self, **data):
+        env_api_key = os.getenv("LLAMA_CLOUD_API_KEY")
+        if env_api_key:
+            data["llama_cloud_api_key"] = env_api_key
+        super().__init__(**data)
+
+    def update_env_api_key(self):
+        """
+        Update the environment variable for the API key.
+        """
+        import dotenv
+
+        if self.llama_cloud_api_key:
+            # Update runtime environment variable
+            os.environ["LLAMA_CLOUD_API_KEY"] = self.llama_cloud_api_key
+            # Update .env file
+            dotenv.set_key(
```

Comment:
reading this, i think it makes sense to encapsulate the API keys that are needed by loaders to the file loader model.
similar we might want to encapsulate env keys for tool configuration in the tool model later.

but one question: how does this interfere with the env config model? we have to ensure that both models (env config and file loader) can be serialized and deserialized independently without impact. e.g. if the env config is serialized we don't want `LLAMA_CLOUD_API_KEY` to be overwritten.

Reply:
Each model just manages the environment variables linked with its fields, so they are independent.